ops/charm.py

```diff
@@ -261,62 +344,157 @@ class RelationCreatedEvent(RelationEvent):
+    determined using no more than the name of the joining unit and the
+    remote `private-address` setting, which is always available when
+    the relation is created and is by convention not deleted.
     """
 
 
 class RelationChangedEvent(RelationEvent):
-    """Represents the `relation-changed` hook from Juju.
-
-    This is triggered whenever there is a change to the data bucket for a related
-    application or unit. Look at `event.relation.data[event.unit/app]` to see the
-    new information.
+    """Event triggered when relation data changes.
+
+    This event is triggered whenever there is a change to the data
+    bucket for a related application or unit. Look at
+    `event.relation.data[event.unit/app]` to see the new information,
+    where `event` is the event object passed to the callback method
+    bound to this event.
+
+    This event always fires once, after :class: `RelationJoinedEvent`
+    event, and will subsequently be fire whenever that remote unit
+    changes its settings for the relation. Callback methods bound to
+    this event should be the only ones that rely on remote relation
+    settings. They should not error if the settings are incomplete,
+    since it can be guaranteed that when the remote unit changes its
+    settings, the event will be fire again.
+
+    The settings that be queried, or set, are determined by the
+    relation’s interface.
     """
 
 
 class RelationDepartedEvent(RelationEvent):
-    """Represents the `relation-departed` hook from Juju.
-
-    This is the inverse of the RelationJoinedEvent, representing when a unit
-    is leaving the relation (the unit is being removed, the app is being removed,
-    the relation is being removed). It is fired once for each unit that is
-    going away.
+    """Event triggered when a unit leaves a relation.
+
+    This is the inverse of the :class: `RelationJoinedEvent`,
+    representing when a unit is leaving the relation (the unit is
+    being removed, the app is being removed, the relation is being
+    removed). It is fired once for each unit that is going away.
+
+    When the remote unit is known to be leaving the relation, this
+    will result in the :class: `RelationChangedEvent` firing at least
+    once, after which the :class: `RelationDepartedEvent` will
+    fire. The :class: `RelationDepartedEvent` will fire once
+    only. Once the :class: `RelationDepartedEvent` has fired no
+    further :class: `RelationChangedEvent` events will fire.
+
+    Callback methods bound to this event may be used to remove all
+    references to the departing remote unit, because there’s no
+    guarantee that it’s still part of the system; it’s perfectly
+    probable (although not guaranteed) that the system running that
+    unit has already shut down.
+
+    Once all callback methods bound to this event have been run for
+    such a relation, the unit agent will fire, the :class:
+    `RelationBrokenEvent` event.
```

Comment:
hmm, have you built the docs to check if it copes with the newline in the middle of that xref?